### PR TITLE
Update parsing and ETL for SCH clinical data files.

### PR DESCRIPTION
**Pull request by davereinhart**
Update parsing for SCH clinical data files to include ICD-10 codes. The
codes are stored in a series of 10-12 new columns in the workbooks and are
converted to an array to be included in JSON output.

### CODE REVIEW

**Comment by davereinhart** [comment]
Related backoffice PR, to be merged after this one:
https://github.com/seattleflu/backoffice/pull/236

**Review by seidelma — APPROVED** [review]
I ran the code against one of the XLS files mentioned in the backoffice PR, as well as an empty one. Looks good to me. One minor complaint: there are some superfluous whitespace changes at lines 206 and 270 that pollute the diff.

**Comment by davereinhart** [comment]
- [ ] Wait for final SCH year 3 file before merging this branch. Additional will be required for new columns.

**Comment by davereinhart** [comment]
[Reprocess Y1, Y2, Y3 SCH data](https://trello.com/c/BRuEXAKV/1205-reprocess-y1-y2-y3-sch-data)

**Comment by davereinhart** [comment]
Requires https://github.com/seattleflu/id3c/pull/273 to be merged first.

**Comment by davereinhart** [comment]
Additional changes made to ingest new columns added to Year 3 file:  COVID vaccine status and date (dose 1 and 2), COVID Vaccine manufacturer, and Flu Shot date.

**Review by seidelma — APPROVED** [review]
Looks good to me, just one change in the `add_icd10()` function.

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 1]
```diff
@@ -238,6 +289,20 @@ def parse_sch(sch_filename, output):
...
+    DataFrame.
+    """
+    def icd10(series: pd.Series) -> pd.Series:
+        """ Returns an array of unique ICD-10 codes from a given *series*. """
+        icd10_columns = [col for col in df.columns if col.startswith('diag_cd')]
+        icd10_codes = [ series[i] for i in icd10_columns if not pd.isna(series[i])]
+        return list(set(icd10_codes))
+
+    df['ICD10'] = df.apply(icd10, axis='columns')
+    return df
```
[the remark this answers is not in the document]
Comment (davereinhart) [reply lib/seattleflu/id3c/cli/command/clinical.py]: Good catch!  I was following the pattern of the other function, but you're right. There are a number of them that I updated which didn't need to return the dataframe.

File: lib/seattleflu/id3c/cli/command/etl/clinical.py [hunk 2]
```diff
@@ -286,6 +304,115 @@ def flu_shot(flu_shot_response: Optional[Any]) -> list:
...
+        LOG.debug("No COVID shot manufacturer name found.")
+        return [None]
+
+    if isinstance(covid_shot_manufacturer_name, str):
+        covid_shot_manufacturer_name = covid_shot_manufacturer_name.lower().strip()
+
+    valid_covid_manufacturers = [
+        "pfizer",
+        "moderna",
+    ]
```
Comment (kschwabefry) [inline lib/seattleflu/id3c/cli/command/etl/clinical.py:371]: Seems like this list might need to expand later, but raising the warning as below should clue us in if we need to update this. :thumb:

File: lib/seattleflu/id3c/cli/command/etl/clinical.py [hunk 3]
```diff
@@ -286,6 +304,115 @@ def flu_shot(flu_shot_response: Optional[Any]) -> list:
...
+    Given a *covid_shot_response*, returns yes/no value for CovidShot key.
+
+    >>> covid_shot(0.0)
+    ['no']
+
+    >>> covid_shot('TRUE')
+    ['yes']
+
+    >>> covid_shot('maybe')
+    Traceback (most recent call last):
```
Comment (kschwabefry) [inline lib/seattleflu/id3c/cli/command/etl/clinical.py:319]: Thanks for adding the doctests here!

**Review by kschwabefry — APPROVED** [review]

**Review by seidelma — APPROVED** [review]
Looks good, thanks Dave!

**Comment by davereinhart** [comment]
Deployed.

**Pull request merged**